lib/Dialect/Secret/Transforms/DistributeGeneric.cpp

```diff
@@ -298,6 +300,7 @@ struct SplitGeneric : public OpRewritePattern<GenericOp> {
            "source and target generics must be in the same block");
 
     IRMapping cloningMp;
+    mlir::DominanceInfo dom(sourceGeneric->getParentOfType<func::FuncOp>());
```

Comment:
Since this is only used for an assert, does this constructor incur any additional runtime penalty? I know computing dominance info can be expensive... If it does, consider putting it inside an `ifdef DEBUG` just before to the assert (see https://github.com/llvm/llvm-project/blob/e8a15eca92f1a10b3af4f4e52f54d9d2d7612bf5/libc/src/assert/assert.h#L17C1-L17C1)